internal/commands/push/command.go

```diff
@@ -126,8 +126,38 @@ func (cmd *Command) Handler(profile *cli.Profile, ui terminal.UI, clients cli.Cl
 		diffs = append(diffs, appDiffs...)
 
 		if cmd.inputs.IncludeDependencies {
-			// TODO(REALMC-8242): diff dependencies better
-			diffs = append(diffs, "+ New function dependencies")
+			dependencies, err := local.FindAppDependencies(app.RootDir)
```

Comment:
so we need to collect diffs for the dependencies before this point (similar to how hosting diffs are handled)

that is because if we find no diffs, then we know we don't need to do anything for this import and return early